pygmt/src/binstats.py

```diff
@@ -0,0 +1,101 @@
+"""
+binstats - Bin spatial data and determine statistics per bin
+"""
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+@fmt_docstring
+@use_alias(
+    C="statistic",
+    E="empty",
+    G="outgrid",
+    I="spacing",
+    N="normalize",
+    R="region",
+    S="search_radius",
+    T="tiling",
+    V="verbose",
+    W="weight",
```

Comment:
Still missing some parameter docstrings, need to copy from https://github.com/GenericMappingTools/gmt/blob/6.3.0/doc/rst/source/gmtbinstats.rst.

- [ ] normalize (N)
- [ ] search_radius (S)
- [ ] tiling (T)
- [ ] weight (W)